web/app.js

```diff
@@ -2046,6 +2046,69 @@ function handleMouseWheel(evt) {
 window.addEventListener('DOMMouseScroll', handleMouseWheel);
 window.addEventListener('mousewheel', handleMouseWheel);
 
+function touchSwipeDetector(evt) {
```

Comment:
Sure. I'll address these review comments in a second commit - let  me know if you'd prefer to have the comments squashed into the base commit.